internal/querynodev2/handlers.go

```diff
@@ -249,6 +251,72 @@ func (node *QueryNode) querySegments(ctx context.Context, req *querypb.QueryRequ
 	}, nil
 }
 
+func (node *QueryNode) optimizeSearchParams(ctx context.Context, req *querypb.SearchRequest, channel string) *querypb.SearchRequest {
+	log := log.Ctx(ctx).With(zap.String("channel", channel))
+	// no hook applied, just return
+	if node.queryHook == nil {
+		return req
+	}
+
+	serializedPlan := req.GetReq().GetSerializedExprPlan()
+	// plan not found
+	if serializedPlan == nil {
+		log.Warn("serialized plan not found")
+		return req
+	}
+
+	channelNum := req.GetTotalChannelNum()
+	// not set, change to conservative channel num 1
+	if channelNum <= 0 {
+		channelNum = 1
```

Comment:
2 may be better, as it's the number of channels by default

Reply:
1 is more conservative (hook will not reduce topk and params too much)